src/Core/Framework/DataAbstractionLayer/Search/Criteria.php

```diff
@@ -115,6 +115,11 @@ class Criteria extends Struct
      */
     public function __construct(array $ids = [])
     {
+        if (\count($ids) === 0) {
```

Comment:
this will also be the case, if I do not set the `$ids` parameter at all, doesn't it?

Reply:
You're right - but how would I distinguish a default `[]` from a provided `[]`? :disappointed:

Reply:
I changed the PR to make this work - I am just not sure if this is the way to go